make/compiler/Makefile.cray-prgenv

```diff
@@ -46,10 +46,19 @@ GEN_DYNAMIC_FLAG = -dynamic
 LIB_STATIC_FLAG =
 LIB_DYNAMIC_FLAG = -shared
 
-# With locale models other than 'flat' we will need libnuma.  That is
-# so far available only in .so form, but the PrgEnv compilers default
-# to forcing static linking.  So, override them in this case.
+#
+# Circumstances may push us to choose dynamic linking instead of the
+# PrgEnv default static linking.
+#
 ifneq ($(CHPL_MAKE_LOCALE_MODEL), flat)
+# With locale models other than 'flat' we will need libnuma.  That is
+# so far available only in .so form, so override them in this case.
+RUNTIME_LFLAGS += -dynamic
+else ifeq ($(CHPL_MAKE_COMM), ofi)
+# libfabric calls dlopen() and getaddrinfo().  With static linking, the
+# references to these cause link-time warnings that the corresponding
+# dynamic libraries will be required at execution time.  With dynamic
+# linking we avoid these warnings.
 RUNTIME_LFLAGS += -dynamic
```

Comment:
Not required for this PR, but we should see what other groups are doing about this. We know dynamic linking on Crays can lead to non-trivial performance regressions, and it seems hard to believe other groups will rely on full dynamic linking for ofi